lib/server.js

```diff
@@ -141,18 +141,27 @@ module.exports = function createServer(config, log) {
     // that we won't block. These are typically for Mozilla QA purposes.
     // They should be checked after everything else so as not to pay the
     // overhead of checking the many requests that are *not* QA-related.
-    if (result.block || result.suspect) {
-      if (ip in allowedIPs.ips || (email && allowedEmailDomains.isAllowed(email))) {
-        log.info({
-          op: 'request.check.allowed',
-          ip: ip,
-          block: result.block,
-          suspect: result.suspect
-        })
-        result.block = false
-        result.suspect = false
-      }
+    if ((result.block || result.suspect) && isWhitelisted(ip, email)) {
```

Comment:
The second test has been pulled out to `isWhitelisted` so that it can be re-used elsewhere.